adam/experiment/live_demo.py

```diff
@@ -0,0 +1,319 @@
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+
+import cv2
+import numpy as np
+import requests
+import yaml
+from vistautils.parameters import Parameters
+
+from adam.curriculum.curriculum_from_files import phase3_process_scene_dir
+from adam.experiment.observer import YAMLLogger
+from adam.learner.integrated_learner import SimulatedIntegratedTemplateLearner
+from adam.paths import (
+    TESTING_CURRICULUM_DIR,
+    LEARNERS_DIR,
+    PREPROCESSING_DIR,
+    TRAINING_CURRICULUM_DIR,
+)
+
+MENU_STR = "ADAM Demo UI Valid Commands: (d)ecode, (e)xit, (r)eset curriculum: "
+DECODE_STR = "Filepath to the image to decode: "
+
+
+def configure_learner(
+    learner_path: Path,
+) -> SimulatedIntegratedTemplateLearner:
+    """Load a learner from a pickle file."""
+    try:
+        return pickle.load(open(learner_path, "rb"))
+    # if the learner can't be loaded, just instantiate the default learner and notify the user
+    except OSError as err:
+        raise RuntimeError(
+            "Unable to load existing learner, exiting demo script"
+        ) from err
+
+
+@dataclass
+class DemoCurriculum:
+    curriculum_name: str
+    curriculum_path: Path
+    situation_number: int = 0
+    color_seed: int = 42
+
+    def initialize_curriculum(self) -> None:
+        """Create the curriculum directory"""
+        if self.curriculum_path.is_dir():
+            print(
+                "Curriculum already exists. Configuring situation number automatically."
+            )
+            info_path = self.curriculum_path / "info.yaml"
+            if info_path.exists():
+                with open(info_path, encoding="utf-8") as info_file:
+                    info_yaml = yaml.safe_load(info_file)
+                self.situation_number = int(info_yaml["num_dirs"])
+
+        self.curriculum_path.mkdir(parents=True, exist_ok=True)
+
+    def reset(self, *, initialize_after_reset: bool = True) -> None:
+        """Reset the curriculum directory, optionally reinitialize it."""
+        if self.curriculum_path.is_dir():
+            self.curriculum_path.rmdir()
+
+        if initialize_after_reset:
+            self.initialize_curriculum()
+
+    def preprocess(
+        self, image_path: Path, host: str, stroke_model_path: Path, stroke_python: Path
+    ) -> Path:
+        """Preprocess an image with stroke extraction to configure the curriculum scene."""
+        situation_dir = self.curriculum_path / f"situation_{self.situation_number}"
+        situation_dir.mkdir(parents=True)
+        situation_img_path = situation_dir / "rgb_0.png"
+        segmentation_path = situation_dir / "semantic_0.png"
+        img = cv2.imread(str(image_path))
+        img2 = None
+        try:
+            img2 = cv2.cvtColor(img, cv2.COLOR_RGBA2RGB)
+            print("Image is RGBA")
+        except ValueError:
+            print("Image is RGB")
+            img2 = img
+        finally:
+            cv2.imwrite(str(situation_img_path), img2)
```

Comment:
(No change needed) It bothers me that I feel like we could do something smarter here... but we don't need to, so 🤷‍♂️.

Reply:
Agreed, happy to call this "live demo code" and not come up with a better solution unless needed.